crawler/crawlutils.py

```diff
@@ -508,6 +509,35 @@ def snapshot(
                     overwrite=overwrite
                 )
 
+	    try:
+                container_event = eventQ.get(block=True, timeout = wait_period)
```

Comment:
yes, timeout is essentially max wait time. it will wait maximum of 5 'wait-period', if no event received it will break out. But, if there's an event within wait-period, it will immediately break out.